app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,86 @@
+class TasksController < ApplicationController
+    def index
+        @tasks = Task.all 
+    end
+
+    def show
+        task_id = params[:id]
+        @task = Task.find_by(id: task_id)
+        if @task.nil?
+            redirect_to tasks_path
+            return
+        end
+    end
+
+    def new
+        @task = Task.new
+    end
+
+    def create
+        @task = Task.new(name: params[:task][:name], description: params[:task][:description], completed_at: params[:task][:completed_at])
+        if @task.save
+            redirect_to task_path(@task)
+            return
+        else
+            render :new
+            return
+        end
+    end
+
+    def edit
+        @task = Task.find_by(id: params[:id])
+
+        if @task.nil?
+            redirect_to tasks_path
+            return
+        end
+    end
+
+    def update
+        @task = Task.find_by(id: params[:id])
+        if @task.nil?
+            redirect_to tasks_path
+            return
+        elsif @task.update(
+            name: params[:task][:name],
+            description: params[:task][:description],
+            completed_at: params[:task][:completed_at]
+        )
+            redirect_to task_path(@task)
+            return
+        else
+            render :edit
+            return
+        end
+    end
+
+    def destroy
+        task_id = params[:id]
+        @task = Task.find_by(id: task_id)
+
+        if @task.nil?
+            head :not_found
+            return
+        end
+
+        @task.destroy
+
+        redirect_to tasks_path
+        return
+    end
+
+    def complete
```

Comment:
The only to unmark a task complete is through the `edit` action. You should add this functionality to a custom method. Consider how making two separate methods, for instance `mark_complete` and `mark_incomplete` would make this action idempotent.